crates/redstone/src/core/config.rs

```diff
@@ -30,3 +35,242 @@ pub struct Config {
     /// for determining whether the data is current in the context of blockchain time.
     pub block_timestamp: TimestampMillis,
 }
+
+impl Config {
+    /// Verifies all members of the config.
+    ///
+    /// This method checks whether all configs members are correct.
+    ///
+    /// # Returns
+    ///
+    /// * Success `()` if config is valid or Err with `Error` otherwise.
+    pub(crate) fn verify(&self) -> Result<(), Error> {
+        self.validate_feed_ids_list()?;
+        self.validate_signers_list()
+    }
+
+    #[inline]
+    fn validate_feed_ids_list(&self) -> Result<(), Error> {
+        self.is_feed_ids_empty()?;
+        has_duplicates(&self.feed_ids)
+            .map_or_else(|| Ok(()), |v| Err(Error::ConfigReocuringFeedId(v)))
+    }
+
+    #[inline(always)]
+    fn is_feed_ids_empty(&self) -> Result<(), Error> {
+        if self.feed_ids.is_empty() {
+            return Err(Error::ConfigEmptyFeedIds);
+        }
+
+        Ok(())
+    }
+
+    #[inline]
+    fn validate_signers_list(&self) -> Result<(), Error> {
+        self.is_signers_count_in_threshold()?;
+        self.is_signer_count_not_exceeded()?;
+        has_duplicates(&self.signers)
+            .map_or_else(|| Ok(()), |v| Err(Error::ConfigReocuringSigner(v)))
+    }
+
+    #[inline(always)]
+    fn is_signers_count_in_threshold(&self) -> Result<(), Error> {
+        if self.signers.len() < self.signer_count_threshold as usize || self.signers.is_empty() {
```

Comment:
the function starting with `is_` should return boolean for me. so here it should be changed to `verify_signer_count`. 

and we don't need (but we can) do it in one function if needed :)